datafusion/sqllogictest/test_files/array.slt

```diff
@@ -6938,13 +6938,13 @@ select range(5),
        range(2, 5),
        range(2, 10, 3),
        range(10, 2, -3),
-       range(1, 5, -1),
-       range(1, -5, 1),
-       range(1, -5, -1),
-       range(DATE '1992-09-01', DATE '1993-03-01', INTERVAL '1' MONTH),
-       range(DATE '1993-02-01', DATE '1993-01-01', INTERVAL '-1' DAY),
-       range(DATE '1989-04-01', DATE '1993-03-01', INTERVAL '1' YEAR),
-       range(DATE '1993-03-01', DATE '1989-04-01', INTERVAL '1' YEAR)
+       range(arrow_cast(1, 'Int8'), 5, -1),
+       range(arrow_cast(1, 'Int16'), arrow_cast(-5, 'Int8'), 1),
+       range(arrow_cast(1, 'Int32'), arrow_cast(-5, 'Int16'), arrow_cast(-1, 'Int8')),
+       range(DATE '1992-09-01', DATE '1993-03-01', arrow_cast('1 MONTH', 'Interval(YearMonth)')),
+       range(DATE '1993-02-01', arrow_cast(DATE '1993-01-01', 'Date64'), INTERVAL '-1' DAY),
+       range(arrow_cast(DATE '1989-04-01', 'Date64'), DATE '1993-03-01', INTERVAL '1' YEAR),
+       range(arrow_cast(DATE '1993-03-01', 'Date64'), arrow_cast(DATE '1989-04-01', 'Date64'), INTERVAL '1' YEAR)
```

Comment:
Uplift tests to ensure coercion is handled properly for different types

Reply:
I recommend making this more explicit by:
1. Revert the  change to the existing tests
2. Add a new query right below this one that adds the casts to the other integer types